client/src/views/ActiveUsersPage.vue

```diff
@@ -79,36 +172,153 @@ import {
   IonPage,
   IonLabel,
   IonListHeader,
+  IonCheckbox,
+  IonText,
 } from '@ionic/vue';
+import {
+  personRemove,
+  personAdd,
+  eye,
+} from 'ionicons/icons';
 import UserListItem from '@/components/Users/UserListItem.vue';
 import UserCard from '@/components/Users/UserCard.vue';
 import ButtonOption from '@/components/ButtonOption.vue';
 import { isAdmin } from '@/common/permissions';
 import ListGridToggle from '@/components/ListGridToggle.vue';
 import { DisplayState } from '@/components/ListGridToggle.vue';
+import { popoverController } from '@ionic/vue';
+import UserContextMenu from '@/components/UserContextMenu.vue';
+import { UserAction } from '@/components/UserContextMenu.vue';
 import ActionBar from '@/components/ActionBar.vue';
+import { MockUser, getMockUsers } from '@/common/mocks';
+import { Ref } from 'vue';
+import { onMounted } from '@vue/runtime-core';
 
 const displayView = ref(DisplayState.List);
-const userList = ['User1', 'User2', 'User3'];
+const userList: Ref<MockUser[]> = ref([]);
+const userListItemRefs: Ref<typeof UserListItem[]> = ref([]);
+const userGridItemRefs: Ref<typeof UserCard[]> = ref([]);
+const allUsersSelected = ref(false);
+const interminateState = ref(false);
 
 const filteredUsers = computed(() => {
-  return userList;
+  const revokedUsers = userList.value.filter((user) => {
+    return user.revoked === false;
+  });
+  return revokedUsers;
+});
+
+const selectedUsersCount = computed(() => {
+  if (displayView.value === DisplayState.List) {
+    return userListItemRefs.value.filter((item) => item.isSelected).length;
+  } else {
+    return userGridItemRefs.value.filter((item) => item.isSelected).length;
+  }
 });
 
 function inviteUser(): void {
   console.log('Invite user clicked');
 }
+
+function viewCommonWorkspace(): void {
+  console.log('View common workspace clicked');
+}
+
+function onUserSelect(_user: MockUser, _selected: boolean): void {
+  console.log(selectedUsersCount.value);
+
+  if (selectedUsersCount.value === 0) {
+    allUsersSelected.value = false;
+    selectAllUsers(false);
+  }
+  // check global checkbox if all users are selected
+  if (selectedUsersCount.value === userList.value.length) {
+    allUsersSelected.value = true;
+  } else {
+    allUsersSelected.value = false;
+  }
+
+  if (selectedUsersCount.value === 0 || selectedUsersCount.value === userList.value.length) {
+    interminateState.value = false;
+  } else {
+    console.log('No users selected');
```

Comment:
Logic is weird here. If `selectedUsersCount.value === 0`, it should mean that no user is selected, therefore you're in the `if`, not in the `else`.

Reply:
I fixed it.